src/main/kotlin/com/beust/kobalt/maven/DependencyManager.kt

```diff
@@ -39,19 +36,13 @@ public class DependencyManager @Inject constructor(val executors: KobaltExecutor
      * Reorder dependencies so that if an artifact appears several times, only the one with the higest version
      * is included.
      */
+    // TODO it may cause problems if there are file dependencies in the list, we can't compare file dependencies with maven dependencies
     public fun reorderDependencies(dependencies: Collection<IClasspathDependency>): List<IClasspathDependency> {
-        val result = arrayListOf<IClasspathDependency>()
-        val map : ArrayListMultimap<String, IClasspathDependency> = ArrayListMultimap.create()
-        // The multilist maps each artifact to a list of all the versions found
-        // (e.g. {org.testng:testng -> (6.9.5, 6.9.4, 6.1.1)}), then we return just the first one
-        dependencies.forEach {
-            map.put(it.shortId, it)
-        }
-        for (k in map.keySet()) {
-            val l = map.get(k)
-            Collections.sort(l, Collections.reverseOrder())
-            result.add(l.get(0))
-        }
-        return result
+        return dependencies.groupBy { it.shortId + ":" + it.classifierIfAvailable() }
```

Comment:
It is impossible to revert this because it is *required* for valid resolution. Will try to mock all this stuff that is required for dependency instance creation.